app/vmalert/config/config.go

```diff
@@ -52,22 +54,43 @@ func (g *Group) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err != nil {
 		return fmt.Errorf("failed to marshal group configuration for checksum: %w", err)
 	}
-	// change default value to prometheus datasource.
-	if g.Type.Get() == "" {
-		g.Type.Set(NewPrometheusType())
-	}
 
 	h := md5.New()
 	h.Write(b)
 	g.Checksum = fmt.Sprintf("%x", h.Sum(nil))
 	return nil
 }
 
-// Validate check for internal Group or Rule configuration errors
+// SetDefaultFields sets some fields with default values if nil
+func (g *Group) SetDefaultFields(defaultGroupInterval time.Duration) {
```

Comment:
I don't think we need to move this code to another method, especially public method. It gives not benefits from my perspective

Reply:
Fixed!